### PR TITLE
Update RTL examples

### LINKED ISSUE #33170: Text placement is not in the correct place in the example checkout-rtl
I was observing the example of [checkout-rtl](https://getbootstrap.com/docs/5.0/examples/checkout-rtl/), in which I saw many errors. If you look at the following images, you notice that the gray text positions in the "rtl" mode is not true, and the gray text is before the black text.

### PR DESCRIPTION
**Pull request by aqeelat**
Hi,

The current RTL examples have many inconsistencies and mistranslations. I'm taking a deeper look and translating them better.
Don't merge yet.

### CODE REVIEW

**Comment by ffoodd** [comment]
Thanks a lot for tackling this, this is highly appreciated!
Please tell us if we wan help or if you have any questions.

**Comment by AqeelAT** [comment]
Do I really have to rebase? it's really creating a bunch of unneeded commits.

**Comment by ffoodd** [comment]
It shouldn't, but if your PR is ready we'll rebase it by ourselves if needed.

**Comment by AqeelAT** [comment]
I think I'm done with all examples, except for the cheatsheet, which would need to be looked at by a fresh pair of eyes. Pinging @aldilaff and @Nuhasami
Should I provide RTL versions of other examples? If so, which ones should I prioritize? I would probably add them as separate PRs though.

File: site/content/docs/5.0/examples/blog-rtl/index.html [hunk 1]
```diff
@@ -195,7 +198,7 @@ <h4 class="fst-italic">في مكان آخر</h4>
 </main><!-- /.container -->
 
 <footer class="blog-footer">
-  <p>تم تصميم نموذج المدونة لـ <a href="https://getbootstrap.com/"> Bootstrap </a> بواسطة <a href="https://twitter.com/mdo">mdo</a>.</p>
+  <p>تم تصميم نموذج المدونة لـ <a href="https://getbootstrap.com/">Bootstrap</a> بواسطة <a href="https://twitter.com/mdo"><bdi lang="en" dir="ltr">@mdo</bdi></a>.</p>
```
Comment (ffoodd) [inline site/content/docs/5.0/examples/blog-rtl/index.html:201]: Snapshoting this in my mind for our future bi-di example templates :heart:

**Comment by ffoodd** [comment]
@AqeelAT We prioritized some examples in our first implementation—based on how much they'd change in RTL— but feel free to add some of the other ones if you feel it makes sense.

In #32330 I added in the "Need help" part a topic regarding bi-directionnal templates. We don't have any of them and I'm not sure how to tackle this, so if you have any insights (or even bi-di contents in the wild) that could help a lot. We really ened some examples in order to document how-to for bi-di layouts since it'd been asked a few times already.

This PR is pushing the overall quality of our RTL support a lot, that's unvaluable so **thanks a lot**!

**Comment by AqeelAT** [comment]
@ffoodd I'm glad I was able to help. As for bi-di, I already used it extensively in the cheatsheet. Check out the tables there. However, I'll try to come up with a dedicated example.

**Comment by AqeelAT** [comment]
Reminder: we would need to recreate the screenshots before the merge.

**Comment by ffoodd** [comment]
Would it make any sense to use eastern Arabic numerals in examples, @AqeelAT?
This is kindly asked by @Mrahmani71 in #33170: [comment](https://github.com/twbs/bootstrap/issues/33170#issuecomment-784019641).

This is the exact kind of details we need help for, since we don't have any Arabic speaker in the team. Your efforts are highly appreciated!

**Comment by AqeelAT** [comment]
@ffoodd Many Arabic speakers use Hindu numerals. I guess it makes sense to use them in the examples. I'll update the examples.

**Comment by ffoodd** [comment]
@AqeelAT Friendly ping here, do you think it's safe to merge or do you see anything else to do—apart from new screenshots, which I'll handle?

**Comment by AqeelAT** [comment]
I'm sorry I was crazy busy in the past couple of weeks. I'll do a final review and update you within 24 hours.

**Comment by ffoodd** [comment]
No worry nor hurry, just wanted to check back if we didn't forget this 🙂

**Comment by AqeelAT** [comment]
Looks good to me. I think it's ready to be merged.

**Comment by AqeelAT** [comment]
And for the Hindu-Arabic numerals comment that was mentioned in another PR of mine, I'm a strong believer that Arabic text should have [Arabic numerals](https://www.wikiwand.com/en/Arabic_numerals), and therefore we should not replace them in these examples.

**Comment by ffoodd** [comment]
Fair enough, thanks a lot!

**Review by ffoodd — APPROVED** [review]

**Pull request merged**